dynamic-tools/zap-extensions/zapwso2jiraplugin/ExtensionJiraIssueCreater.java

```diff
@@ -0,0 +1,199 @@
+        String BASE_URL = jiraBaseURL;
+
+        log.info("Backup Folder Path" + backupFolder);
+
+        String version = MavenBuildReceiver.getPomVersion(new File(workspace + "/pom.xml"));
+
+        String summary = "[JENKINS][" + product + "-" + version + "][DynamicScan-Nightly-Build-Report]";
+
+        //String summary = "[JENKINS][" + product + "][DynamicScan-Nightly-Build-Report]";
+
+        XmlDomParser xmlParser = new XmlDomParser();
+
+        //Checking if there is any issues exists in the ZAP report generated after the scan
+        boolean issueExist = xmlParser.isIssueExistsInReport();
+
+        //Checking if there is any already available issues reported in the jira
+        String issueKey = xmlParser.checkForIssueExistence(auth, BASE_URL, summary, projectKey);
+
+        String fileNewPath = xmlParser.reNameFile(product, filePath);
+
+        if (issueExist) {
+            try {
+                if (issueKey.equals("")) {
+                    String issueToBeCreated = xmlParser
+                            .createNewTicket(projectKey, asssignee, issueLabel, summary, product);
+                    issue = jira.invokePostMethod(auth, BASE_URL + "/rest/api/2/issue", issueToBeCreated);
+
+                    JSONObject createdIssue = new JSONObject(issue);
+                    issueKey = createdIssue.getString("key");
+                    log.info("Issue is created under : " + issueKey);
+
+                    jira.invokePostComment(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/comment",
+                            xmlParser.createComment());
+
+                    jira.invokePutMethodWithFile(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/attachments",
+                            xmlParser.compressFile(fileNewPath));
+                    log.info("Issue " + issueKey + " is updated with an attachment");
+                } else {
+
+                    updateJiraAttachments.modifyJiraContents(auth, BASE_URL, issueKey);
+
+                    jira.invokePostComment(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/comment",
+                            xmlParser.createComment());
+                    log.info("Issue " + issueKey + " is updated with an attachment");
+
+                    jira.invokePutMethodWithFile(auth, BASE_URL + "/rest/api/2/issue/" + issueKey + "/attachments",
+                            xmlParser.compressFile(fileNewPath));
+
+                }
+            } catch (AuthenticationException e) {
+                log.info("Authentication failed");
+            }
+
+            if (!(" ".equals(backupFolder)))
+                this.moveAttachmentToBackupFolder(backupFolder, xmlParser.compressFile(fileNewPath));
+            else
+                log.info("Back up location is unspecified");
+
+        } else {
+            log.info("There are no issues exists in the report");
+        }
+    }
+
+    public void moveAttachmentToBackupFolder(String path, String filePath) {
```

Comment:
Shouldn't this be a private/protected method? If this is a utility method, please move to a utils class.